subsys/bluetooth/audio/pacs.c

```diff
@@ -246,13 +248,42 @@ static int set_available_contexts(uint16_t contexts, uint16_t *available,
 		return 0;
 	}
 
+	err = k_work_reschedule(&available_contexts_work, PAC_NOTIFY_TIMEOUT);
+	if (err < 0) {
+		return err;
+	}
+
 	*available = contexts;
 
-	err = k_work_reschedule(&available_contexts_work, PAC_NOTIFY_TIMEOUT);
+	return 0;
+}
+
+static int set_supported_contexts(uint16_t contexts, uint16_t *supported,
+				  uint16_t *available)
+{
+	int err;
+
+	if (*supported == contexts) {
+		return 0;
+	}
+
+	err = k_work_reschedule(&supported_contexts_work, PAC_NOTIFY_TIMEOUT);
 	if (err < 0) {
 		return err;
 	}
 
+	*supported = contexts;
+
+	/* Update available contexts if needed*/
+	if ((contexts & *available) != *available) {
+		*available = *available & contexts;
+		err = k_work_reschedule(&available_contexts_work,
+					PAC_NOTIFY_TIMEOUT);
+		if (err < 0) {
+			LOG_WRN("Update available contexts notify failed: %d", err);
+		}
+	}
+
```

Comment:
BAP requires that we always support the `BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED` context type.

We can either require it to be set in `contexts` here, or always just set the bit, but we need to ensure that the resulting context contains the bit.

Reply:
Is the newest revision an appropriate solution?